src/components/Form/Signup/UserInfo.js

```diff
@@ -174,8 +199,37 @@ class UserInfo extends React.Component {
     };
 
     SendPhoneCode = () => {
-        if (this.state.phone_code_sending) return;
         const { intl, locale } = this.props;
+        // check username first
+        if (!this.state.valid_username) {
+            this.props.form.setFields({
+                username: {
+                    value: '',
+                    errors: [
+                        new Error(
+                            intl.formatMessage({ id: 'error_username_invalid' })
+                        ),
+                    ],
+                },
+            });
+            return;
+        }
+        // check email second
+        if (!this.state.valid_email) {
```

Comment:
why to check email in the SendPhoneCode progress? I think these are two separate events.